### PR TITLE
RHTAP-1495: AppSet for RH users configurations

### PR DESCRIPTION
**Pull request by gbenhaim**
Add a new application set that will generate application for RH each tenant defined in https://github.com/redhat-appstudio/tenants-config Each tenant has a directory with the name if his/hers namespace. The generate application will be associated with the "tenants-config" project that restricts which resources can be deployed, to which namespaces and clusters. The project is defined using app-interface.

### CODE REVIEW

File: argo-cd-apps/overlays/production/kustomization.yaml
Comment (Michkov) [inline argo-cd-apps/overlays/production/kustomization.yaml:9]: does not exist

**Review by Michkov — DISMISSED** [review]

File: argo-cd-apps/overlays/production/kustomization.yaml
Comment (gbenhaim) [reply argo-cd-apps/overlays/production/kustomization.yaml:9]: thanks, fixed.

File: argo-cd-apps/base/tenants-config/tenants-config.yaml
Comment (Michkov) [inline argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: should be pinned to specific version

Comment (gbenhaim) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: We  don't want to pin it to specific version. We want ArgoCD to track main for this application.

File: argo-cd-apps/base/tenants-config/tenants-config.yaml
Comment (ernesgonzalez33) [inline argo-cd-apps/base/tenants-config/tenants-config.yaml:32]: Why do we need unlimited retries?

Comment (gbenhaim) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml]: fixed

File: argo-cd-apps/base/tenants-config/tenants-config.yaml
Comment (Michkov) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: None of pull-requests in that project has been reviewed before merging, so I do not trust that project.

Comment (Michkov) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: (but not blocking this PR)

My understading is that we do not want to have some random unreviewed changed, so that's why we are always pinning to specific sha.

CC: @sbose78 @arewm

Comment (gbenhaim) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: pr will get reviewd on https://github.com/redhat-appstudio/tenants-config
I just added a simple directory structure to it so we can get things rolling.

**Review by scoheb — APPROVED** [review]

File: argo-cd-apps/base/tenants-config/tenants-config.yaml
Comment (scoheb) [reply argo-cd-apps/base/tenants-config/tenants-config.yaml:9]: new PR with required review: https://github.com/redhat-appstudio/tenants-config/pull/4

**Pull request merged**
